### ISSUE TITLE
Panic in `bug` when parsing malformed file

### ISSUE THREAD

**Issue by ghost** [sile/libflate#64]
```rust
#[test]
fn test_bug() {
    let input = b"\x04\x04\x04\x05:\x1az*\xfc\x06\x01\x90\x01\x06\x01";
    let mut decoder = Decoder::new(&input[..]);
                                                                        
    let result = io::copy(&mut decoder, &mut io::sink());
}
```

```rust
thread 'deflate::decode::tests::test_bug' panicked at 'bug', src/huffman.rs:124:43
```

**Comment by sile** [comment]
Thank you for creating this issue.
I'll start investigating this problem within a week.
BTW, there are multiple "Decoder" structs in `libflate` ( https://docs.rs/libflate/latest/libflate/?search=decoder ). So could you give me the module name of the above `Decoder` struct?

**Comment by ghost** [comment]
Ah, my bad. I put that test in `deflate/decode.rs`, it's the DEFLATE decoder.

**Comment by sile** [comment]
Thank you!

**Comment by sile** [comment]
Fixed in #65. Thanks again for reporting this problem.
